craft_cli/messages.py

```diff
@@ -439,7 +439,18 @@ def stop(self) -> None:
         if not TESTMODE:
             self.spinner.stop()
         if self.unfinished_stream is not None:
-            print(flush=True, file=self.unfinished_stream)
+            # With unfinished_stream set, the prv_msg object is valid.
+            if self.prv_msg.ephemeral:  # type: ignore
+                # If the last printed message is of 'ephemeral' type, the stop
+                # request must clean and reset the line.
+                cleaner = " " * (_get_terminal_width() - 1)
```

Comment:
I think that the `- 1` does not correspond here, because you're not actually showing the cursor at the end of the terminal (in its last column), because of the second CR.

Reply:
True.

Just note that we are technically "cleaning" one character more than what would ever be written by an ephemeral message. If you are happy with the inconsistency, then we can clean the entire line here.

Reply:
yes, it's fine, because the ephemeral message actually uses one more position that our cleanup here (the cursor)